stdlib/public/core/Span/Span.swift

```diff
@@ -0,0 +1,792 @@
+  ///     must be greater or equal to zero, and less than `count`.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @unsafe
+  @_alwaysEmitIntoClient
+  public subscript(unchecked position: Index) -> Element {
+    //FIXME: change to unsafeRawAddress or unsafeAddress when ready
+    _read {
+      let elementOffset = position &* MemoryLayout<Element>.stride
+      let address = _start().advanced(by: elementOffset)
+      let binding = Builtin.bindMemory(
+        address._rawValue, 1._builtinWordValue, Element.self
+      )
+      defer { Builtin.rebindMemory(address._rawValue, binding) }
+      yield UnsafePointer<Element>(address._rawValue).pointee
+    }
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 6.1, *)
+extension Span where Element: BitwiseCopyable {
+
+  /// Accesses the element at the specified position in the `Span`.
+  ///
+  /// - Parameter position: The offset of the element to access. `position`
+  ///     must be greater or equal to zero, and less than `count`.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public subscript(_ position: Index) -> Element {
+    get {
+      _precondition(
+        UInt(bitPattern: position) <  UInt(bitPattern: _count),
+        "index out of bounds"
+      )
+      return self[unchecked: position]
+    }
+  }
+
+  /// Accesses the element at the specified position in the `Span`.
+  ///
+  /// This subscript does not validate `position`; this is an unsafe operation.
+  ///
+  /// - Parameter position: The offset of the element to access. `position`
+  ///     must be greater or equal to zero, and less than `count`.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @unsafe
+  @_alwaysEmitIntoClient
+  public subscript(unchecked position: Index) -> Element {
+    get {
+      let elementOffset = position &* MemoryLayout<Element>.stride
+      let address = _start().advanced(by: elementOffset)
+      return address.loadUnaligned(as: Element.self)
+    }
+  }
+}
+
+//MARK: sub-spans
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 6.1, *)
+extension Span where Element: ~Copyable {
+
+  /// Constructs a new span over the items within the supplied range of
+  /// positions within this span.
+  ///
+  /// The returned span's first item is always at offset 0; unlike buffer
+  /// slices, extracted spans do not share their indices with the
+  /// span from which they are extracted.
+  ///
+  /// - Parameter bounds: A valid range of positions. Every position in
+  ///     this range must be within the bounds of this `Span`.
+  ///
+  /// - Returns: A `Span` over the items within `bounds`
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _extracting(_ bounds: Range<Index>) -> Self {
```

Comment:
What `@lifetime` do these operations set on their return value?

Reply:
Feel free to be explicit, we're pretty confident that `@lifetime(self)` is a good default in most cases.